src/main/java/net/minecraftforge/event/item/ItemEvent.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2018.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.item;
+
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.item.Item;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.fml.common.eventhandler.Event;
+
+/**
+ * ItemEvent is fired when an event involving any Item occurs.<br>
+ * If a method utilizes this {@link Item} as its parameter, the method will
+ * receive every child event of this class.<br>
+ * <br>
+ * {@link #item} contains the entity that caused this event to occur.<br>
+ * <br>
+ * All children of this event are fired on the {@link MinecraftForge#EVENT_BUS}.<br>
+ **/
+public class ItemEvent extends Event
+{
+    private final Item item;
+
+    public ItemEvent(Item item) {
+        this.item = item;
+    }
+
+    public Item getItem() {
+        return item;
+    }
+
+    @HasResult
+    public static class InCreativeTab extends ItemEvent {
```

Comment:
should probably be made a standalone class as there currently is no other usecase for a `Item`-based `ItemEvent` class